pkg/reconciler/handlers.go

```diff
@@ -12,3 +14,33 @@ func Handler(h func(interface{})) cache.ResourceEventHandler {
 		DeleteFunc: h,
 	}
 }
+
+type ErrorHandler interface {
```

Comment:
Could we invert the dependency here? A retrying NetworkErrorReconciler that takes wraps a reconciler interface and is constructed with a reconciler?

Reply:
I like it, let me give it a shot 😄